csv_export_partner/models/csv_export_partner.py

```diff
@@ -55,24 +55,24 @@ class PartnerCSVExport(models.TransientModel):
     _filename_template = "CLI_%Y%m%d_%H%M_%S%f.csv"
 
     def get_recordset(self):
-        # cf csv_export_invoice.py
-        exported_invoices = self.env["account.invoice"].search(
-            [
-                ("state", "!=", "draft"),
-                ("state", "!=", "cancel"),
+        invoice_domain = [("state", "!=", "draft"), ("state", "!=", "cancel")]
+        payment_domain = [("journal_id.type", "=", "cash"), ("state", "!=", "draft")]
+        if self.manual_date_selection:
+            invoice_domain += [
                 ("date", ">=", self.start_date),
                 ("date", "<", self.end_date),
             ]
-        )
-        # cf csv_export_payment.py
-        exported_payments = self.env["account.payment"].search(
-            [
-                ("journal_id.type", "=", "cash"),
-                ("state", "!=", "draft"),
+            payment_domain += [
                 ("payment_date", ">=", self.start_date),
                 ("payment_date", "<", self.end_date),
             ]
-        )
+        else:
+            exported_domain = ("export_to_sftp", "=", False)
+            invoice_domain.append(exported_domain)
+            payment_domain.append(exported_domain)
+
+        exported_invoices = self.env["account.invoice"].search(invoice_domain)
+        exported_payments = self.env["account.payment"].search(payment_domain)
```

Comment:
@victor-champonnois thinking again about this, I think this is not correct : if invoices are exported before the partners, they will be marked as "exported" and no partners will show up in this filter.

This custom filter on partners was created because 
- the filter on partner write date returned a lot of partners (all those who were touched by an update or a compute function)
- the filter on partner create date did not work because we want the updates on partners

The options I see are :

1) make sure partners are exported first
2) export partners based on _recently_ exported invoiced based on `export_to_sftp` (datetime field)
    - _recently_ needs to be decided with Jérémie
3) export all partners not exported linked to an invoice
    - no updates of partners

What do you think ? Do you think about other options ?